x-pack/plugins/alerting/server/maintenance_window_client/get_maintenance_window_date_and_status.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import { DateRange, MaintenanceWindowStatus } from '../../common';
+
+export interface DateSearchResult {
+  event: DateRange;
+  index: number;
+  status: MaintenanceWindowStatus;
+}
+
+export interface MaintenanceWindowDateAndStatus {
+  eventStartTime: string | null;
+  eventEndTime: string | null;
+  status: MaintenanceWindowStatus;
+  index?: number;
+}
+
+// Returns the most recent/relevant event and the status for a maintenance window
+export const getMaintenanceWindowDateAndStatus = ({
```

Comment:
it looks like this function is not used any where, is that correct?

Reply:
It's being used in `getMaintenanceWindowFromRaw`, basically we use this helper to decorate the maintenance window SO with the current `status`, the `event_start/end_dates` when returned by the API, kind of similar to `getAlertFromRaw`